alf/algorithms/trac_algorithm.py

```diff
@@ -112,20 +113,29 @@ def predict(self, time_step: ActionTimeStep, state, epsilon_greedy):
     def rollout(self, time_step: ActionTimeStep, state, mode):
         """Rollout for one step."""
         policy_step = self._ac_algorithm.rollout(time_step, state, mode)
-        return policy_step._replace(
-            info=TracInfo(
-                observation=time_step.observation,
-                state=self._ac_algorithm.convert_train_state_to_predict_state(
-                    state),
-                ac=policy_step.info))
+        return self._replace_info(policy_step, time_step, state)
 
     def train_step(self, exp: Experience, state):
         policy_step = self._ac_algorithm.train_step(exp, state)
+        return self._replace_info(policy_step, exp, state)
+
+    def _replace_info(self, policy_step, time_step, state):
```

Comment:
We can make ddpg also include action_distribution in info structure so we can avoid doing this here.

Reply:
> We can make ddpg also include action_distribution in info structure so we can avoid doing this here.

I did this in a different PR #321